examples/developer-tools/src/output-generators/javascript_definition_generator.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as Blockly from 'blockly';
+import {JavascriptGenerator} from 'blockly/javascript';
+
+export class JavascriptDefinitionGenerator extends JavascriptGenerator {
+  /**
+   * Scrub handles concatenating "next" connection blocks.
+   * This is a copy of the regular scrub_ function for JS, but
+   * we also add newlines between next blocks.
+   *
+   * @override
+   */
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  scrub_(block: Blockly.Block, code: string, thisOnly = false): string {
+    let commentCode = '';
+    // Only collect comments for blocks that aren't inline.
+    if (!block.outputConnection || !block.outputConnection.targetConnection) {
+      // Collect comment for this block.
+      let comment = block.getCommentText();
+      if (comment) {
+        comment = Blockly.utils.string.wrap(comment, this.COMMENT_WRAP - 3);
+        commentCode += this.prefixLines(comment + '\n', '// ');
+      }
+      // Collect comments for all value arguments.
+      // Don't collect comments for nested statements.
+      for (let i = 0; i < block.inputList.length; i++) {
+        if (block.inputList[i].type === Blockly.inputs.inputTypes.VALUE) {
+          const childBlock = block.inputList[i].connection.targetBlock();
+          if (childBlock) {
+            comment = this.allNestedComments(childBlock);
+            if (comment) {
+              commentCode += this.prefixLines(comment, '// ');
+            }
+          }
+        }
+      }
```

Comment:
I think this can be rewritten as follows:

```suggestion
      for (let i = 0; i < block.inputList.length; i++) {
        if (block.inputList[i].type !== Blockly.inputs.inputTypes.VALUE) continue;
        const childBlock = block.inputList[i].connection.targetBlock();
        if (!childBlock) continue;
        comment = this.allNestedComments(childBlock);
        if (!comment) continue;
        commentCode += this.prefixLines(comment, '// ');
      }
```

(And indeed a similar tranformation can be applied WRT the enclosing `if (!block.outputConnection…` too.)

Reply:
I agree, but I took this directly out of the `scrub_` function in core, and I think it's worth leaving them the same so it's easier to compare and see what changed. But I think I will play with adding the newlines directly to the code instead of in scrub, since I'm going to filter out empty lines later anyway. Then I won't need to override this function at all. So I'll come back to this!